### PR TITLE
Backport of code from V13 for mail_message

### PR DESCRIPTION
**Pull request by PierreCourt**
Description of the issue/feature this PR addresses:

Perf issues in mail_init_messaging, solved by backport of the V13 code

OPW_2281836_1

First run is with code of v12, second with code of V13

Here is the svg of the flamegraph for v12 code : https://drive.google.com/file/d/1SfPvI5zIB6tD02Uxq5h51ft2JHSNTDQF/view?usp=sharing
Here is the svg of the flamegraph for V13 code : https://drive.google.com/file/d/12tqjEsaOBAq1jN-38BNJdmh8LA7b8RF_/view?usp=sharing

The test was done with 892 records.

Current behavior before PR: 
Slow

Desired behavior after PR is merged:
Faster

Linked to the ticket : [2281836](https://www.odoo.com/web?debug#action=333&active_id=49&id=2281836&menu_id=4720&model=project.task&view_type=form)

### CODE REVIEW

**Comment by rim-odoo** [comment]
@PierreCourt Can you add the ticket number to the PR description? Thanks :)

**Comment by PierreCourt** [comment]
@rim-odoo I added it to the description :)

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -531,36 +532,40 @@ def message_format(self):
...
-                'module_icon': '/mail/static/src/img/smiley/mailfailure.jpg',
-                'notifications': dict((notif.res_partner_id.id, (notif.email_status, notif.res_partner_id.name)) for notif in message.notification_ids.sudo())
-            }
+            notifications = msgid_to_notif[message.id]
+            if not any(notification.is_email for notification in notifications):
+                continue
+            info = dict(message._get_mail_failure_dict(),
+                        notifications=dict(
+                            (notif.res_partner_id.id, (notif.email_status, notif.res_partner_id.name)) for notif
+                            in notifications))
```
Comment (nim-odoo) [inline addons/mail/models/mail_message.py:554]: keep the indent of the original commit so it's easier to compare both versions.

**Review by nim-odoo — COMMENTED** [review]
The commit message must be rewritten to match our standards. Moreover, it is useful to link the original commit 488a5d5af5418ed5c223af5b61aea0b3100508c1

Other than that I'm not against it, but I didn't test nor carefully reviewed.

**Comment by nim-odoo** [comment]
fw-bot up to saas-12.3

**Comment by C3POdoo** [comment]
Dear @PierreCourt,

Thank you for your contribution but the version 12.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**